### DISCUSSION TITLE
Supabase Storage API row level security issue when uploading object

### DISCUSSION THREAD

**Discussion by riefer02** [supabase/supabase#9274] [Questions] [answered]
Hello Supabase and friends,

I am hitting the following API route attempting to add a file to my storage bucket:

URL:
```
${supabaseAPI}/storage/v1/object/avatars/${avatarFile.name}
```
method: `'POST'`
headers:
```
{
  Authorization: `Bearer ${supabaseKey}`,
  "Content-Type": "multipart/form-data",
}
```

body:
The body of the request contains the `avatarFile` File to be uploaded

On my Deno instance I get a `400` error `Bad Request`

In my supabase admin storage logs I get and `Error Object` with a message of:

```
"message": "new row violates row-level security policy for table \"objects\"",
```

I am using my project API key from the API settings I assume this API KEY will allow me to insert/upload files to the storage bucket.

Is there something I need to configure before the storage API with the token allows me to insert new rows and not violate a row policy?

Appreciate any help in this matter. Cheers 😄

**Comment by GaryAustin1** [comment, marked as answer]
This discusses the policy rules you need to set for storage.  https://supabase.com/docs/guides/storage#add-security-rules

For upload https://supabase.com/docs/reference/javascript/storage-from-upload#notes you need an insert policy set.

If you use your service_role key in your function it should bypass RLS.
